### DISCUSSION TITLE
What is the Typical Approach for Multiple Replica Sets when Creating Microsoft.AAD/DomainServices

### DISCUSSION THREAD

**Discussion by brianstringfellow** [Azure/ResourceModules#1409] [Q&A]
What is the typical approach for multiple replica sets when creating Microsoft.AAD/DomainServices?

I was looking at the following ARM template to see if there was a different approach or something I have missed when setting up Azure Active Directory Domain Services. https://github.com/Azure/ResourceModules/blob/main/arm/Microsoft.AAD/DomainServices/deploy.bicep

In my case, I have four hubs configured in a mesh network in different regions. I have replica sets in each of the hubs. I realized early on that I could not deploy my ARM template with all four regions when setting up the environments for the first time. The domain services error indicated that the initial deployment could contain only one replica set and I can add only a single replica set each time. My approach ended up consisting of a procedural execution of PowerShell both before and after the ARM template deployment.

Upon the very first deployment of domain services, the PowerShell script detects that domain services do not yet exist and passes a parameter with a single replica set. After the ARM template deploys with the single replica set, additional PowerShell script runs to deploy the remaining three replicas sets, one by one.

Subsequent deployments of domain services, the PowerShell script detects the domain services and how many replica sets are present. The PowerShell then passes in exact existing number of replica sets.

Reference document:
https://docs.microsoft.com/en-us/azure/active-directory-domain-services/concepts-replica-sets#:~:text=managed%20domain%20with%20three%20replica%20sets

**Comment by AlexanderSehr** [comment]
Hey @mararam, would you any an input for this question? Unfortunately, I cannot give any advice here and would need to give it a spin myself.
